vars/goBuildAndRelease.groovy

```diff
@@ -0,0 +1,163 @@
+    body()
+
+    toolsNode(toolsImage: 'stakater/pipeline-tools:1.6.0') {
+        container(name: 'tools') {
+            withCurrentRepo(type: 'go') { def repoUrl, def repoName, def repoOwner, def repoBranch ->
+                String chartPackageName = ""
+                String srcDir = WORKSPACE
+                def kubernetesDir = WORKSPACE + "/deployments/kubernetes"
+
+                def chartTemplatesDir = kubernetesDir + "/templates/chart"
+                def chartDir = kubernetesDir + "/chart"
+                def manifestsDir = kubernetesDir + "/manifests"
+
+                def dockerContextDir = WORKSPACE + "/build/package"
+                def dockerImage = repoOwner.toLowerCase() + "/" + repoName.toLowerCase()
+                def dockerImageVersion = ""
+                def binary = "./build/package/" + repoName.toLowerCase()
+
+                // Slack variables
+                def slackChannel = "${env.SLACK_CHANNEL}"
+                def slackWebHookURL = "${env.SLACK_WEBHOOK_URL}"
+
+                def utils = new io.fabric8.Utils()
+                def git = new io.stakater.vc.Git()
+                def helm = new io.stakater.charts.Helm()
+                def templates = new io.stakater.charts.Templates()
+                def common = new io.stakater.Common()
+                def chartManager = new io.stakater.charts.ChartManager()
+                def docker = new io.stakater.containers.Docker()
+                def stakaterCommands = new io.stakater.StakaterCommands()
+                def slack = new io.stakater.notifications.Slack()
+                try {
+                    stage('Export dockerImage') {
+                        sh """
+                          export DOCKER_IMAGE=${dockerImage}
+                        """
+                    }
+
+                    stage('Download Dependencies') {
+                        sh """
+                            cd ${srcDir}
+                            make install
+                        """
+                    }
+
+                    stage('Run Tests') {
+                        sh """
+                            cd ${srcDir}
+                            make test
+                        """
+                    }
+
+                    stage('Build Binary') {
+                        sh """
+                            export BINARY=${binary}
+                            cd ${srcDir}
+                            make build
+                        """
+                    }
+
+                    if (utils.isCI()) {
+                        stage('CI: Publish Dev Image') {
+                            dockerImageVersion = stakaterCommands.getBranchedVersion("${env.BUILD_NUMBER}")
+                            def builder = "docker.io/" + "${dockerImage}:${dockerImageVersion}"
+                            sh """
+                              export DOCKER_TAG=${dockerImageVersion}
+                              export BUILDER=${builder}
```

Comment:
use the env `DOCKER_IMAGE` instead of `BUILDER` 
as the binary docker image which we want to push is created by the name  in `DOCKER_IMAGE`